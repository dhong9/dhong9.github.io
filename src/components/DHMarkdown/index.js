import ReactMarkdown from "react-markdown";
import Typography from "@mui/material/Typography";
import Box from "@mui/material/Box";

const components = {
  components: {
    h1: ({ ...props }) => <Typography gutterBottom={true} variant="h4" component="h1" {...props} />,
    h2: ({ ...props }) => <Typography gutterBottom={true} variant="h6" component="h2" {...props} />,
    h3: ({ ...props }) => <Typography gutterBottom={true} variant="subtitle1" {...props} />,
    h4: ({ ...props }) => (
      <Typography gutterBottom={true} variant="caption" paragraph={true} {...props} />
    ),
    p: ({ ...props }) => <Typography paragraph={true} {...props} />,
    a: "@mui/material/Link",
    li: ({ ...props }) => <Box component="li" sx={{ mt: 1, typography: "body1" }} {...props} />,
  },
};

export default function DHMarkdown(props) {
  return <ReactMarkdown components={components} {...props} />;
}
