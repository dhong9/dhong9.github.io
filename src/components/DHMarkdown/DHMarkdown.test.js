// React testing libraries
import renderer from "react-test-renderer";

// Component to test
import DHMarkdown from "components/DHMarkdown";

describe("DHMarkdown", () => {
  it("renders h1 component", () => {
    const component = renderer.create(<DHMarkdown># Hello world!</DHMarkdown>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders h2 component", () => {
    const component = renderer.create(<DHMarkdown>## Hello world!</DHMarkdown>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders h3 component", () => {
    const component = renderer.create(<DHMarkdown>### Hello world!</DHMarkdown>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders h4 component", () => {
    const component = renderer.create(<DHMarkdown>#### Hello world!</DHMarkdown>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders p component", () => {
    const component = renderer.create(<DHMarkdown>Hello world!</DHMarkdown>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a component", () => {
    const component = renderer.create(<DHMarkdown>[Hello world](https://example.com)</DHMarkdown>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
