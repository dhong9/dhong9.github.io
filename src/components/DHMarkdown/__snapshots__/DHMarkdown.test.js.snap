// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DHMarkdown renders a component 1`] = `
<p>
  <a
    href="https://example.com"
  >
    Hello world
  </a>
</p>
`;

exports[`DHMarkdown renders h1 component 1`] = `
<h1>
  Hello world!
</h1>
`;

exports[`DHMarkdown renders h2 component 1`] = `
<h2>
  Hello world!
</h2>
`;

exports[`DHMarkdown renders h3 component 1`] = `
<h3>
  Hello world!
</h3>
`;

exports[`DHMarkdown renders h4 component 1`] = `
<h4>
  Hello world!
</h4>
`;

exports[`DHMarkdown renders p component 1`] = `
<p>
  Hello world!
</p>
`;
